src/materialized/src/mux.rs

```diff
@@ -7,51 +7,43 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+//! Methods common to servers listening for TCP connections.
+
 use std::sync::Arc;
 
 use async_trait::async_trait;
 use futures::stream::{Stream, StreamExt};
-use tokio::io::{self, AsyncWriteExt};
+use tokio::io::{self};
 use tokio::net::TcpStream;
-use tracing::{debug, error};
+use tracing::error;
 
-use mz_ore::netio::{self, SniffedStream, SniffingStream};
 use mz_ore::task;
 
 use crate::http;
 
-type Handlers = Vec<Box<dyn ConnectionHandler + Send + Sync>>;
-
-/// A mux routes incoming TCP connections to a dynamic set of connection
-/// handlers. It enables serving multiple protocols over the same port.
-///
-/// Connections are routed by sniffing the first several bytes sent over the
-/// wire and matching them against each handler, in order. The first handler
-/// to match the connection will be invoked.
-pub struct Mux {
-    handlers: Handlers,
-}
```

Comment:
getting rid of this seems right, we probably want to rename the file though!

Reply:
Ah we probably want to get rid of the file entirely. I think it's fine to leave as is in this PR and then I'll swing by and get rid of it entirely in a fast follow. #12891 tracks that work.